Add ODIN SDK

svenpaulsen opened the pull request:
### SteamDB app page links to a few games using this

- https://steamdb.info/app/2479810/

### Brief explanation of the change

ODIN is a cross-platform SDK engineered to seamlessly integrate real-time voice chat into multiplayer games, applications, and websites. This PR allows finding its files.

`php tests/GenerateTestStrings.php` was used to generate the test file.

SteamTracker commented:
After running changed rules in this pull request *(this bot only tests rules.ini changes)*:

Changed technologies: **SDK.ODIN**

These apps now match: **(18)** [672890](https://steamdb.info/app/672890/depots/) [785770](https://steamdb.info/app/785770/depots/) [910800](https://steamdb.info/app/910800/depots/) [1220860](https://steamdb.info/app/1220860/depots/) [1480080](https://steamdb.info/app/1480080/depots/) [1601920](https://steamdb.info/app/1601920/depots/) [1890650](https://steamdb.info/app/1890650/depots/) [1957780](https://steamdb.info/app/1957780/depots/) [2003160](https://steamdb.info/app/2003160/depots/) [2009460](https://steamdb.info/app/2009460/depots/) [2018820](https://steamdb.info/app/2018820/depots/) [2077180](https://steamdb.info/app/2077180/depots/) [2195750](https://steamdb.info/app/2195750/depots/) [2227830](https://steamdb.info/app/2227830/depots/) [2331150](https://steamdb.info/app/2331150/depots/) [2437390](https://steamdb.info/app/2437390/depots/) [2479810](https://steamdb.info/app/2479810/depots/) [2651270](https://steamdb.info/app/2651270/depots/)

Please confirm that all these changed apps are correct.

svenpaulsen commented:
> After running changed rules in this pull request _(this bot only tests rules.ini changes)_:
> 
> Changed technologies: **SDK.ODIN**
> 
> These apps now match: **(18)** [672890](https://steamdb.info/app/672890/depots/) [785770](https://steamdb.info/app/785770/depots/) [910800](https://steamdb.info/app/910800/depots/) [1220860](https://steamdb.info/app/1220860/depots/) [1480080](https://steamdb.info/app/1480080/depots/) [1601920](https://steamdb.info/app/1601920/depots/) [1890650](https://steamdb.info/app/1890650/depots/) [1957780](https://steamdb.info/app/1957780/depots/) [2003160](https://steamdb.info/app/2003160/depots/) [2009460](https://steamdb.info/app/2009460/depots/) [2018820](https://steamdb.info/app/2018820/depots/) [2077180](https://steamdb.info/app/2077180/depots/) [2195750](https://steamdb.info/app/2195750/depots/) [2227830](https://steamdb.info/app/2227830/depots/) [2331150](https://steamdb.info/app/2331150/depots/) [2437390](https://steamdb.info/app/2437390/depots/) [2479810](https://steamdb.info/app/2479810/depots/) [2651270](https://steamdb.info/app/2651270/depots/)
> 
> Please confirm that all these changed apps are correct.

Confirmed! 😇

Lifeismana commented on rules.ini:
```diff
@@ -274,6 +274,7 @@ NVIDIA_Texture_Tools = (?:^|/)nvtt(?:_64)?\.dll$
 NVIDIA_TXAA = (?:^|/)GFSDK_TXAA
 NWJS = (?:^|/)nw\.(?:dll|pak)$
 OculusXRPlugin = (?:^|/)OVRPlugin\.dll$
+ODIN = (?:^|/)(?:lib)?odin\.(?:dylib|dll|so)$
```
```suggestion
ODIN = (?:^|/)(?:lib)?odin\.(?:dll|so)$
```
looking for macos libs is not needed since it seems unlikely to me that there will ever be a macos only game published on steam (or at least in a near futur) (& all apps are detected without this part)

(don't forget to fix your test strings to take that change in account)

Lifeismana reviewed (changes requested):
on top of this change, can you also add some non matching tests (check [here](https://github.com/SteamDatabase/FileDetectionRuleSets/blob/main/CONTRIBUTING.md#contributing) for an example)

svenpaulsen commented:
Thanks for the feedback! I've removed the macOS libraries from the pattern as suggested by @Lifeismana, updated the test file and added some non matching tests.

Lifeismana commented:
Looks good to me, thank you!

svenpaulsen commented:
Any idea when this will go live?
